### DISCUSSION TITLE
Option to delay landscaping until X years passed

### DISCUSSION THREAD

**Discussion by Shituation** [OpenTTD/OpenTTD#10187] [Features]
To disable trolling by joining a game, using the starting money + loans + many companies to cause havok on the map.

**Comment by Eddi-z** [comment]
There is no technical solution to trolling. The only way to stop trolls is permanent moderation.

**Reply by Shituation** [reply]
So your solution is not having more options and staying the same. Either you are one of those trolls or I don't get it. I honestly don't understand people like you in real life either.

**Reply by JGRennison** [reply]
Running an open public server is not a good idea if you want any kind of in game quality. You'd be better off passwording your server and possibly not including it in the public server list, and doing the filtering before people even join the server.

**Reply by Shituation** [reply]
I don't need advice to not use game features or weak justification on why "not change anything". I'm not asking what to do with trolls. I'm asking a new feature that can be used or not by users. Thanks.

**Reply by Eddi-z** [reply]
> So your solution is not having more options and staying the same.

Well, suppose we implement your suggestion, the trolls will find a different way to troll within a day... it's an endless game of whack-a-mole either way.

how much effort are you willing to put into this arms race, and possibly cause collateral damage at any step on the way?

**Comment by PikkaBird** [comment]
Config settings to limit the maximum landscaping allowed in a given timeframe already exist:

terraform_per_64k_frames = 4194304
terraform_frame_burst = 4096
clear_per_64k_frames = 4194304
clear_frame_burst = 4096

I don't see why the setting you suggest couldn't also be added, but (like any other feature) it's not going to happen until someone who wants it (ie, you) does the work.

**Reply by Shituation** [reply]
I don't understand those settings to be honest, can you explain those settings a little to know how to adjust them to my game? thanks

**Reply by Eddi-z** [reply]
the "burst" setting tells how many tiles you can clear/terraform in one step, and once you used up the burst, it slowly fills back up at the speed set by the other setting

64k frames is a time period of around 2 years

**Reply by FLHerne** [reply]
It's documented (a bit confusingly) in [section 4.1 of the multiplayer docs.](https://github.com/OpenTTD/OpenTTD/blob/master/docs/multiplayer.md#41-imposing-landscaping-limits)

64k frames is just over half an hour in real-world time, or longer if the game is lagging.

**Reply by Shituation** [reply]
~~What do the "clear" options do?
Sorry to bother but the documentation seems to be made only for the person that has written it. No direct reference to the settings and also this is not my mother tongue.~~

EDIT: I just realized the clear option is the "TNT". Why is it mixed with terraforming? 🙃

**Reply by Eddi-z** [reply]
because of other forms of griefing.

**Reply by LC-Zorg** [reply]
These settings are quite effective in limiting attempts at trolling and deforming the map. The main problem with using them is that these settings cannot be changed or reset when you want to continue playing offline.

An alternative solution is to significantly increase the cost of terraforming. Such solutions are used on OpenTTD-Polska servers where the cost of changing one terrain point is higher or only slightly less than the maximum loan (£228k vs. £500k/£200k). This means that no new player, especially a weak one, can immediately freely transform the terrain and thus annoy other players. The advantage can also be that thanks to the high costs, more advanced players have something to do with the accumulated money later in the game.
The problem with this solution is that it cannot be used in a pure game without add-ons > Using even one add-on makes the server invisible to many new players > thus the server attendance decreases. Often to zero > and such servers are usually skipped because no one wants to play online alone.

**Comment by andythenorth** [comment]
Ok, so I join a server, but landscaping is disabled for X years, so now I can't play the game? 😄 

Why? 😄

**Reply by Shituation** [reply]
> Ok, so I join a server, but landscaping is disabled for X years, so now I can't play the game? 😄
> 
> Why? 😄

You can play as anyone else, finding a good spot for your stations. Nobody needs landscaping at start because its expensive; and if you, like me, make a rather flat map, its not really needed at start. The only use that can be given at start is to troll and go. Remember this is an option not a permanent change, server hosters like me can make use of this anti-troll patch.

**Reply by nielsmh** [reply]
You're just moving the griefing potential around.
A player joining the game part-way through will be unable to modify the landscape, and thus have an even bigger disadvantage against the players already in the game.
Additionally, the players already in the game will be able to grief against the new player, by doing simple landscape modifications getting in the way of the new player's construction, and the new player will have no way to counter it.

**Reply by Shituation** [reply]
> You're just moving the griefing potential around.

So we do nothing about it. Nice. Let the trolls do as they please, do not interfere.😕

> A player joining the game part-way through will be unable to modify the landscape, and thus have an even bigger disadvantage against the players already in the game.

Like anybody joining an already started game? your limit there is the money and landscaping is expensive. 

> Additionally, the players already in the game will be able to grief against the new player

Look, that argument is simply not valid. Any player in any game with power (money) can bully other players if they want as things are now. I also can't see the relation between not having a terrain tool with your statement, what would you do if you had it and some powerful player decides to trash your company with the tool? how much can you counter the bullies if they want to bully you? (remember: being new to the game, low money)

My "feature" only offloads the admin work from kids with nothing better to do. No balancing needed. No gameplay mechanics affected in any serious way. On top of that, server rules can be shown as I do, telling players this. If they find this setting so bad, they can decide to not join the game. Simple.

This are all excuses for not touching the code like always. You don't want to do the job and are here to stop people from asking upgrades to the game. Its fine, you do it for free, but at least don't come around posts trying your best to desestimate user's ideas with no real arguments. Not answering anymore since this is all a game to try to desestimate the feature, and since you will desestimate it, I'm not losing any more time with this.

**Reply by nielsmh** [reply]
It's a game of whack-a-mole, if you block one avenue of griefing, bored people will find new ways to do it.

As it has already been said, you can't solve social problems with technology.
But seriously, try the landscaping limits options already described to you. As a bonus, they can also prevent established players from deleting entire mountains and such.
Or, you could use a NewGRF to adjust the landscaping costs.

Your "no gameplay mechanics affected" is actually false though. As I say, new players will be at an _even bigger_ disadvantage if they can't landscape. Existing players already have an upper hand because they're existing, established companies. New players will have even fewer options to make optimal routes to have a chance to compete, if they have to run around hills, take pointless grades, can't flatten land to build an airport, and can't prepare a canal for a ship route.

**Reply by Eddi-z** [reply]
> You don't want to do the job 

This is your fundamental mistake. This is a hobby project. It's nobody's "job".

**Comment by James103** [comment]
GS API idea:

Expose the landscaping, tile clearing, tree planting, and object construction allowances to the GS API, so that a Game Script can read and set the current value of those allowances for each company.

This should allow much more control over who can landscape and in what amounts and under what conditions, instead of just a flat "you may clear up to X tiles at once and a maximum of Y tiles every 64k frames".

For example, a Game Script can allow bigger, more established companies to do more landscaping than new or small companies.
Example: new companies get 16 tiles burst and 4 tiles/day, but this increases to 64 tiles burst and 16 tiles/day when the company is more established (by some metric such as value, age, and/or performance rating).

**Comment by LC-Zorg** [comment]
> Eddi-z: There is no technical solution to trolling. The only way to stop trolls is permanent moderation.
> nielsmh: It's a game of whack-a-mole, if you block one avenue of griefing, bored people will find new ways to do it.

I think you overestimate the intelligence and stubbornness of the vast majority of trolls. Besides, even with the more intelligent ones, if you take away their favorite, readily available toys, trolling becomes less attractive. You're right that there's no way to completely eliminate such incidents, but that doesn't mean that it should abandon all efforts and opportunities to make game hosting less of a nuisance and despite everything reduce the amount of abuses. Just because a game has 100 bugs doesn't mean you shouldn't fix one of them. Receiving a dozen or so reports every day and calling an admin for a situation that could be significantly reduced or even eliminated can be really discouraging. I don't think it's just shifting the problem. The fence, even if it is very full of holes, will still be better than nothing.
